src/stream-tcp-reassemble.c

```diff
@@ -1838,6 +1838,7 @@ int StreamTcpReassembleHandleSegment(ThreadVars *tv, TcpReassemblyThreadCtx *ra_
         dir = UPDATE_DIR_BOTH;
     }
 
+    const bool reversed_before_ack_handling = (p->flow->flags & FLOW_DIR_REVERSED) != 0;
     /* handle ack received */
     if ((dir == UPDATE_DIR_OPPOSING || dir == UPDATE_DIR_BOTH) &&
         StreamTcpReassembleHandleSegmentUpdateACK(tv, ra_ctx, ssn, opposing_stream, p) != 0)
```

Comment:
All this code can be put in the if block `if ((dir == UPDATE_DIR_OPPOSING || dir == UPDATE_DIR_BOTH)`, can it not ?